universal-application-tool-0.0.1/app/assets/javascripts/main.ts

```diff
@@ -159,11 +159,35 @@ function attachLineClampListeners() {
   applicationCardDescriptions.forEach(el => el.addEventListener("click", removeLineClamp));
 }
 
+function attachModalListeners() {
+  const modalContainer = document.querySelector('#modal-container');
+  const modalGlassPane = document.querySelector('#modal-glass-pane');
+  // Find and connect each modal to its button
+  const modals = Array.from(document.querySelectorAll('.cf-modal'));
+  modals.forEach(modal => {
+    const modalButton = document.querySelector(`#${modal.id}-button`);
+    modalButton.addEventListener("click", function() {
+      modalContainer.classList.toggle('hidden');

```

Comment:
You probably don't need to worry about hiding the container, glassPane, and the modal here. Since there can be multiple hidden modals, you'll have to hide whichever one is visible, and you'll have to hide the container, but not the pane.

Alternatively, you could store the non-visible modals in a separate hidden div and just overwrite the content of modal when you need to show a new one. The only problem with that approach is you'll experience a bad UI redraw if you're chaining modals, but it does mean that you can have only one close button.

Reply:
The glass pane is always shown, but the modal container is hidden. And the close button closes only the modal it is part of.